pkg/pmem-csi-driver/pmem-csi-driver.go

```diff
@@ -251,24 +189,23 @@ func (csid *csiDriver) Run() error {
 	csid.gatherers = append(csid.gatherers, cmm.GetRegistry())
 
 	switch csid.cfg.Mode {
-	case Controller:
-		rs := registryserver.New(csid.clientTLSConfig, csid.cfg.DriverName)
-		csid.gatherers = append(csid.gatherers, rs.GetMetricsGatherer())
-		cs := NewMasterControllerServer(rs)
-
-		if err := s.Start(csid.cfg.Endpoint, nil, cmm, ids, cs); err != nil {
-			return err
+	case Webhooks:
+		namespace := os.Getenv("POD_NAMESPACE")
+		if namespace == "" {
+			return errors.New("POD_NAMESPACE env variable is not set")
 		}
-		if err := s.Start(csid.cfg.RegistryEndpoint, csid.serverTLSConfig, nil /* no metrics gathering for registry at the moment */, rs); err != nil {
-			return err
+		// Just the scheduler extender!
+		if csid.cfg.schedulerListen == "" {
+			return errors.New("webhooks mode needs a scheduler listen address")
 		}
-
-		// Also run scheduler extender?
-		if csid.cfg.schedulerListen != "" {
-			c := scheduler.CapacityViaRegistry(rs)
-			if _, err := csid.startScheduler(ctx, cancel, c); err != nil {
-				return err
-			}
+		factory := informers.NewSharedInformerFactoryWithOptions(csid.cfg.client, resyncPeriod,
+			informers.WithNamespace(namespace),
+		)
+		podLister := factory.Core().V1().Pods().Lister()
+		c := scheduler.CapacityViaMetrics(namespace, csid.cfg.DriverName, podLister)
+		factory.Start(ctx.Done())
+		if _, err := csid.startScheduler(ctx, cancel, c); err != nil {
+			return err
 		}
 	case Node:
```

Comment:
We can move identity server creation from https://github.com/intel/pmem-csi/blob/7b7f17953f9a9cb79bb754e4e5bf94d4c9c1076d/pkg/pmem-csi-driver/pmem-csi-driver.go#L169-L172 to here.

Reply:
Done. I also removed the unused error return value to make it simpler and consistent.

Reply:
Done.